src/ir/ssa_names_test.cc

```diff
@@ -22,34 +22,35 @@
 namespace raksha::ir {
 namespace {
 
+template <typename T>
 class SsaNamesTest : public ::testing::Test {
  public:
   SsaNamesTest() : test_op_("test") {}
 
  protected:
   Operator test_op_;
+  T first_entity_;
+  T second_entity_;
 };
 
-TEST_F(SsaNamesTest, GetOrCreateIDReturnsUniqueIDsForOperations) {
-  SsaNames names;
-  Operation first_operation(nullptr, test_op_, {}, {});
-  Operation second_operation(nullptr, test_op_, {}, {});
-  SsaNames::ID first_operation_id = names.GetOrCreateID(first_operation);
-  SsaNames::ID second_operation_id = names.GetOrCreateID(second_operation);
+// Specialize constructor for Operation as it has no default constructor to
+// initialize `first_entity_` and `second_entity_`.
+template <>
+SsaNamesTest<Operation>::SsaNamesTest()
+    : test_op_("test"),
+      first_entity_(nullptr, test_op_, {}, {}),
+      second_entity_(nullptr, test_op_, {}, {}) {}
 
-  EXPECT_NE(first_operation_id, second_operation_id);
-  EXPECT_EQ(names.GetOrCreateID(first_operation), first_operation_id);
-}
+using MyTypes = ::testing::Types<Operation, Block, Module>;
+TYPED_TEST_SUITE(SsaNamesTest, MyTypes);
```

Comment:
I think a more specific alias than `MyTypes` is called for here, that's a pretty generic name. Maybe `SsaTestTypes`? Alternatively, you could just inline the definition of the `MyTypes` alias into the one place where it is used.

Reply:
Done.